src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -848,176 +848,124 @@ public ListenableFuture<DescribeIndexStatsResponse> describeIndexStats(Struct fi
-     *      *                  the default namespace is used.
-     * @param prefix The prefix with which vector IDs must start to be included in the response.
-     * @param paginationToken The token to paginate through the list of vector IDs.
-     * @param limit The maximum number of vector IDs you want to retrieve.
-     * @return {@link ListResponse} containing the list of vector IDs fetched from the specified namespace.
-     *         The response includes vector IDs up to {@code 100} items.
-     * @throws RuntimeException if there are issues processing the request or communicating with the server.
-     *         This includes network issues, server errors, or serialization issues with the request or response.
-     */
-    @Override
-    public ListenableFuture<ListResponse> list(String namespace, String prefix, String paginationToken, int limit) {
-        validateListEndpointParameters(namespace, prefix, paginationToken, limit, true, true, true, true);
-        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setPrefix(prefix).
-                setPaginationToken(paginationToken).setLimit(limit).build();
-        return asyncStub.list(listRequest);
+        /**
+         * Sets the pagination token to retrieve the next page of vector IDs.
+         * @param paginationToken The pagination token to retrieve the next page of vector IDs.
+         * @return The {@link AsyncIndex.ListRequestBuilder} object.
+         */
+        public ListRequestBuilder setPaginationToken(String paginationToken) {
+            this.paginationToken = paginationToken;
+            return this;
+        }
+
+        /**
+         * Sets the limit of vector IDs to retrieve.
+         * @param limit The limit of vector IDs to retrieve.
+         * @return The {@link AsyncIndex.ListRequestBuilder} object.
+         */
+        public ListRequestBuilder setLimit(Integer limit) {
+            this.limit = limit;
+            return this;
+        }
+
+        /**
+         * Builds the {@link ListRequest} object to send to the list endpoint in order to retrieve vector IDs back from an index.
+         * @return The {@link ListenableFuture} object.
+         */
+        public ListenableFuture<ListResponse> build() {
+            // Logic to validate parameters
+            validateListEndpointParameters(namespace, prefix, paginationToken, limit,
+                    namespace != null, prefix != null, paginationToken != null, limit != null);
+
+            // Build and execute the list request
+            ListRequest.Builder builder = ListRequest.newBuilder();
+            if (namespace != null) builder.setNamespace(namespace);
+            if (prefix != null) builder.setPrefix(prefix);
+            if (paginationToken != null) builder.setPaginationToken(paginationToken);
+            if (limit != null) builder.setLimit(limit);
+            ListRequest listRequest = builder.build();
```

Comment:
Given that we already supposedly validated these fields, I'm not sure we actually need to be checking nulls here?

We should be able to chain things like so without caring what was passed after running the validation:
```suggestion
            ListRequest.Builder builder = ListRequest.newBuilder()
                .setNamespace(namespace)
                .setPrefix(prefix)
                .setPaginationToken(paginationToken)
                .setLimit(limit)
                .build()
```